p2p-tests/lib/p2p.ts

```diff
@@ -0,0 +1,304 @@
+// Copyright © 2021 The Radicle Upstream Contributors
+//
+// This file is part of radicle-upstream, distributed under the GPLv3
+// with Radicle Linking Exception. For full terms see the included
+// LICENSE file.
+
+import fetch, { Headers, Request, Response } from "node-fetch";
+import { strict as strictAssert } from "assert";
+
+if (!globalThis.fetch) {
+  globalThis.fetch = fetch;
+  globalThis.Headers = Headers;
+  globalThis.Request = Request;
+  globalThis.Response = Response;
+}
+
+import * as ProxyClient from "../../proxy-client";
+import * as fs from "fs-extra";
+import * as path from "path";
+import * as stream from "stream";
+import * as util from "util";
+import execa from "execa";
+import { StringDecoder } from "string_decoder";
+
+const PADDING = 12;
+const RESET = "\x1b[0m";
+
+const BLUE = "\x1b[34m";
+const CYAN = "\x1b[36m";
+const GREEN = "\x1b[32m";
+const MAGENTA = "\x1b[35m";
+const RED = "\x1b[31m";
+
+const colors = [BLUE, CYAN, GREEN, MAGENTA, RED];
+const assignedColors = {};
+
+const ROOT_PATH = path.join(__dirname, "..", "..");
+const P2P_TEST_PATH = path.join(ROOT_PATH, "p2p-tests");
+const BIN_PATH = path.join(ROOT_PATH, "target", "debug");
+
+const processes = [];
+
+export function cleanUpProcesses() {
+  processes.forEach(node => node.kill("SIGKILL"));
+}
+
+export function workspacePath(paths?: string[]) {
+  return path.join(ROOT_PATH, "p2p-tests", "workspace", ...paths);
+}
+
+export function keyPath(filename: string) {
+  return path.join(P2P_TEST_PATH, "keys", filename);
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function assertEqual(message: string, got: any, expected: any) {
+  log(`ASSERT: ${message}`);
+  strictAssert.deepStrictEqual(got, expected);
+  log("got what was expected:");
+  prettyPrint(got);
+}
+
+function prefix(pfx: string) {
+  if (assignedColors[pfx] === undefined) {
+    assignedColors[pfx] = colors.pop();
+  }
+
+  return `${RESET + assignedColors[pfx] + pfx.padEnd(PADDING) + RESET} | `;
```

Comment:
Can we use [`chalk`](https://www.npmjs.com/package/chalk) for colors?

Can you also explain in a comment why we have the `RESET` at the beginning?

Reply:
76865e44a92871ea6befb4428abb18405f3161ab
4fde3223c93f69a3a7d24b2a86d32046feaf3284